subsys/net/lib/http/http_server.c

```diff
@@ -0,0 +1,1235 @@
+		}
+
+		memset(ctx_client->buffer, 0, sizeof(ctx_client->buffer));
+		ctx_client->offset = 0;
+		ctx_client->server_state = HTTP_SERVER_PREFACE_STATE;
+
+	} else {
+		data = content_200;
+		len = sizeof(content_200);
+
+		enum http_method method = parser.method;
+		const char *method_str = http_method_str(method);
+
+		LOG_DBG("HTTP Method: %s", method_str);
+
+		if (strncmp(method_str, "GET", 3) == 0 && strncmp(url_buffer, "/results", 8) == 0) {
+			handle_get_request(ctx_server, ctx_client->client_fd);
+		} else if (strncmp(method_str, "POST", 4) == 0 &&
+			   strncmp(url_buffer, "/add", 4) == 0) {
+			char *json_start = strstr(ctx_client->buffer, "\r\n\r\n");
+
+			if (json_start) {
+				handle_post_request(ctx_server, json_start);
+			}
+
+			char response[] = "HTTP/1.1 200 OK\r\nContent-Type: "
+					  "text/plain\r\n\r\nAdded successfully.\n";
+			sendall(ctx_client->client_fd, response, strlen(response));
+		} else if (strncmp(method_str, "GET", 3) == 0 && strncmp(url_buffer, "/", 1) == 0) {
+			sprintf(http_response,
+				"HTTP/1.1 200 OK\r\n"
+				"Content-Type: text/html\r\n"
+				"Content-Encoding: gzip\r\n"
+				"Content-Length: %d\r\n\r\n",
+				len);
+			if (sendall(ctx_client->client_fd, http_response, strlen(http_response)) <
+			    0) {
+				close_client_connection(ctx_server, client_index);
+			} else {
+
+				if (sendall(ctx_client->client_fd, data, len) < 0) {
+					LOG_ERR("sendall failed");
+					close_client_connection(ctx_server, client_index);
+				}
+			}
+		} else {
+			const char *not_found_response = "HTTP/1.1 404 Not Found\r\n"
+							 "Content-Length: 9\r\n\r\n"
+							 "Not Found";
+			if (sendall(ctx_client->client_fd, not_found_response,
+				    strlen(not_found_response)) < 0) {
+				close_client_connection(ctx_server, client_index);
+			}
+		}
+		close_client_connection(ctx_server, client_index);
+		memset(ctx_client->buffer, 0, sizeof(ctx_client->buffer));
+		ctx_client->offset = 0;
+	}
+
+	return 0;
+}
+
+int handle_http_done(struct http_server_ctx *ctx_server, struct http_client_ctx *ctx_client,
+		     int client_index)
+{
+	LOG_DBG("HTTP_SERVER_DONE_STATE");
+
+	close_client_connection(ctx_server, client_index);
+
+	return -errno;
+}
+
+int handle_http_frame_headers(struct http_client_ctx *ctx_client)
+{
+	LOG_DBG("HTTP_SERVER_FRAME_HEADERS");
+
+	unsigned char response_headers_frame[16];
+
+	struct http_frame *frame = &ctx_client->current_frame;
+
+	print_http_frames(ctx_client);
+
+	if (ctx_client->offset < frame->length) {
+		return -EAGAIN;
+	}
+
+	const char *method;
+	const char *path;
+
+	if (!has_upgrade_header) {
+		enum http_method method_h2c = parser.method;
+
+		method = http_method_str(method_h2c);
+		path = url_buffer;
+	} else {
+		method = http_hpack_parse_header(ctx_client, HTTP_SERVER_HPACK_METHOD);
+		path = http_hpack_parse_header(ctx_client, HTTP_SERVER_HPACK_PATH);
+	}
+
+	if (strcmp(method, "GET") == 0 && strcmp(path, "/") == 0) {
```

Comment:
again, why do we have a hard-coded application-specific path in the generic http server code? 

See "routing" comments earlier